semgrep-core/src/fixing/Autofix_printer.ml

```diff
@@ -130,18 +128,24 @@ let add_fix_pattern_ast_nodes (tbl : ast_node_table) ast =
   visitor ast
 
 let make_external_printer ~metavars ~target_contents ~fix_pattern_ast
-    ~fix_pattern : AST_generic.any -> Immutable_buffer.t option =
+    ~fix_pattern : AST_generic.any -> (Immutable_buffer.t, string) result =
   let tbl : ast_node_table = ASTTable.create 8 in
   add_metavars tbl metavars;
   add_fix_pattern_ast_nodes tbl fix_pattern_ast;
   fun any ->
-    let* node = ASTTable.find_opt tbl any in
-    let* str =
-      match node with
+    let/ node =
+      ASTTable.find_opt tbl any
```

Comment:
Nit: I don't think it would be that many more characters to do this with a normal match, and it would have been a bit easier for me to read